enterprise/server/remote_execution/uffd/uffd.go

```diff
@@ -162,14 +178,21 @@ func (h *Handler) receiveSetupMsg(ctx context.Context) (*setupMessage, error) {
 }
 
 func (h *Handler) handle(ctx context.Context, memoryStore *blockio.COWStore) error {
+	h.wg.Add(1)
+	defer h.wg.Done()
```

Comment:
nit: this works, but instead of using a wait group it would be slightly simpler / more idiomatic to use a `quit` channel

```go
defer close(h.quit)
```

then in `Stop()`:

```go
<-h.quit
```

I realized that in `setupUFFDHandler` we have a bug though -

```go
	h, err := uffd.NewHandler()
	if err != nil {
		return status.WrapError(err, "create uffd handler")
	}
	c.uffdHandler = h
	sockAbsPath := filepath.Join(c.getChroot(), uffdSockName)
	if err := h.Start(ctx, sockAbsPath, c.memoryStore); err != nil {
		return status.WrapError(err, "start uffd handler")
	}
	return nil
```

we should move `c.uffdHandler = h` to after the `Start()` block, so that we don't call `c.uffdHandler.Stop()` in `Remove()` unless the handler was started.

Reply:
Modified to use a quit channel and moved setting the uffdHandler

Also added an extra check to Stop() so that it won't break if we do happen to call it without calling Start() first